relay-server/src/endpoints/common.rs

```diff
@@ -82,49 +80,6 @@ pub enum BadStoreRequest {
     EventRejected(DiscardReason),
 }
 
-impl BadStoreRequest {
-    fn to_outcome(&self) -> Option<Outcome> {
-        Some(match self {
-            BadStoreRequest::UnsupportedProtocolVersion(_) => {
-                Outcome::Invalid(DiscardReason::AuthVersion)
-            }
-
-            BadStoreRequest::EmptyBody => Outcome::Invalid(DiscardReason::NoData),
-            BadStoreRequest::EmptyEnvelope => Outcome::Invalid(DiscardReason::EmptyEnvelope),
-            BadStoreRequest::InvalidJson(_) => Outcome::Invalid(DiscardReason::InvalidJson),
-            BadStoreRequest::InvalidMsgpack(_) => Outcome::Invalid(DiscardReason::InvalidMsgpack),
-            BadStoreRequest::InvalidMultipart(_) => {
-                Outcome::Invalid(DiscardReason::InvalidMultipart)
-            }
-            BadStoreRequest::InvalidMinidump => Outcome::Invalid(DiscardReason::InvalidMinidump),
-            BadStoreRequest::MissingMinidump => {
-                Outcome::Invalid(DiscardReason::MissingMinidumpUpload)
-            }
-            BadStoreRequest::InvalidEnvelope(_) => Outcome::Invalid(DiscardReason::InvalidEnvelope),
-
-            BadStoreRequest::QueueFailed(event_error) => match event_error {
-                QueueEnvelopeError::TooManyEnvelopes => Outcome::Invalid(DiscardReason::Internal),
-            },
-            BadStoreRequest::ProjectFailed(project_error) => match project_error {
-                ProjectError::FetchFailed => Outcome::Invalid(DiscardReason::ProjectState),
```

Comment:
Interestingly, this was mapped to `DiscardReason::ProjectState` while it was mapped to `DiscardReason::Internal` in the envelope manager's future. There's little point in differentiating between those failure modes in outcomes, so I went with "internal" just like everywhere else, allowing to simplify logic.